blueprint/core/blueprint_ReactApplicationRoot.h

```diff
@@ -394,6 +420,9 @@ namespace blueprint
             JUCE_ASSERT_MESSAGE_THREAD
             jassert(handler);
 
+            // TODO: Here we have a case of `invoke` where the callback is not a temporary. Perhaps
+            // we want two different `invoke` methods: one for immeidately cleaning up temporaries,
+            // and one which persists native functions. `pinvoke`? `invokeWithPersistentNativeFunctions`?
             return engine.invoke("__BlueprintNative__.subscribe", eventType, std::move(handler));
```

Comment:
Funnily enough for "reasons" I've ended up not using this and instead have registered a `__MyApp__.raiseReques`.

Reply:
Ah interesting! Do you think we should back this 2-way event emitter out?

Reply:
Yeah I'm not sure. I guess it's a more "fluxy" architecture if the user has to register specific actions. It may be worth it. I might have been guilty of suggesting the wrong thing!